clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -162,13 +162,13 @@ else if (this.subscriptionType != type)
             throw new IllegalStateException(SUBSCRIPTION_EXCEPTION_MESSAGE);
     }
 
-    public synchronized boolean subscribe(Set<String> topics, ConsumerRebalanceListener listener) {
+    public synchronized boolean subscribe(Set<String> topics, Optional<ConsumerRebalanceListener> listener) {
```

Comment:
Another suggestion is to not change the interface, then we do:

registerRebalanceListener(Optional.ofNullabe(listener)) - this prevents us to do a requireNonNull check.

Reply:
Actually - is listener actually optional here? It seems like if the listener is null, it should complain?

Reply:
That's similar to how I had it before, but we _do_ want to throw an exception if the user passes in a `null` to the `KafkaConsumer.subscribe()` methods. So the `null` check is handled in the `KafkaConsumer` and converted to an `Optional` when we call the `SubscriptionState`.